rs/nns/governance/src/proposals/install_code.rs

```diff
@@ -102,21 +106,21 @@ impl InstallCode {
             format!("InstallCode proposal invalid because of {}", reason),
         )
     }
-}
 
-impl CallCanister for InstallCode {
-    fn canister_and_function(&self) -> Result<(CanisterId, &str), GovernanceError> {
-        let canister_id = self.valid_canister_id()?;
-        if canister_id == ROOT_CANISTER_ID {
-            // TODO(NNS1-3190): support changing root canister
-            return Err(Self::invalid_proposal_error(
-                "InstallCode for root canister is not supported yet",
-            ));
-        }
-        Ok((ROOT_CANISTER_ID, "change_nns_canister"))
+    fn payload_to_upgrade_root(&self) -> Result<Vec<u8>, GovernanceError> {
+        let stop_upgrade_start = !self.skip_stopping_before_installing.unwrap_or(false);
+        let wasm_module = vec![];
```

Comment:
Uh. Where code?? I assume you meant to use self.wasm_module, but forgot?? How can the assert on line 378 be passing??

Reply:
Nice catch! I had the implementation working before, but I removed it to make sure that the state machine test fails without it (and it does), and forgot to put it back... Line 378 does also fail with this version. Fixed now.